luigi/contrib/htcondor.py

```diff
@@ -0,0 +1,370 @@
+python runner.py $BASE $BASE {tarball_param}
+
+if [ -f $BASE/post.sh ]; then
+  . $BASE/post.sh
+fi
+
+"""
+
+
+def _build_job_description(job_params):
+    from classad import ClassAd
+    submit_params = DEFAULT_JOB_PARAMETERS.copy()
+    submit_params.update(job_params)
+    return ClassAd(submit_params)
+
+
+def _copy_script_to_tmp_dir(script, tmp_dir, dst_file_name=""):
+    if script and os.path.exists(script):
+        file_name = os.path.basename(
+            script) if not dst_file_name else dst_file_name
+        tmp_script = os.path.join(tmp_dir, file_name)
+        shutil.copy(script, tmp_script)
+
+
+class HTCondorJobTask(luigi.Task):
+    n_cpu = luigi.IntParameter(default=1, significant=False)
+    memory = luigi.IntParameter(
+        default=300,
+        significant=False, description="Amount of memory to be requested"
+    )
+    universe = luigi.Parameter(default='vanilla', significant=False)
+    copies = luigi.IntParameter(
+        default='1', significant=False,
+        description='How many copies of this task to submit as a bundle')
+    run_locally = luigi.BoolParameter(
+        significant=False,
+        description="run locally instead of on the cluster")
+    poll_time = luigi.IntParameter(
+        significant=False, default=10,
+        description=(
+            "specify time between queries to condor scheduler for the job "
+            "status")
+        )
+    dont_remove_tmp_dir = luigi.BoolParameter(
+        significant=False,
+        description="don't delete the temporary directory (for debugging)")
+    no_tarball = luigi.BoolParameter(
+        significant=False,
+        description="don't tarball (and extract) the luigi project files")
+    base_tmp_dir = luigi.Parameter(
+        default='/tmp', significant=False,
+        description="base location for the temporary job logs")
+    pre_script = luigi.Parameter(
+        default='',
+        significant=False,
+        description="shell script to be run before the payload"
+    )
+    post_script = luigi.Parameter(
+        default='',
+        significant=False,
+        description="shell script to be run after the payload"
+    )
+    additional_hooks = luigi.DictParameter(
+        default={},
+        significant=False,
+        description=(
+            "Additional htcondor hooks to be used for job submission "
+            "(expert-only - read the htcondor documentation).\n"
+            "Example: "
+            "--additional_hooks '{\"+AccountingGroup\": \"group_physics\"}'"
+        )
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(HTCondorJobTask, self).__init__(*args, **kwargs)
+
+    def _fetch_task_failures(self):
+        if not os.path.exists(self.errfile):
+            logger.info('No error file')
+            return []
+        with open(self.errfile, "r") as f:
+            errors = f.readlines()
+        return errors
+
+    def _init_local(self):
+        # Set up temp folder (trim to max filename length)
```

Comment:
Why this strange approach to creating temporary directories?  Why not `mkdtemp` instead?